src/content/content.ts

```diff
@@ -301,9 +301,11 @@ export class ContentHandler {
 
     this.touchClickTracker.onTouchClick = (event: MouseEvent) => {
       // Ignore clicks on interactive elements
+      const possibleTargetElem = event.target as Element | null;
       if (
-        event.target instanceof HTMLAnchorElement ||
-        event.target instanceof HTMLButtonElement
+        possibleTargetElem != null &&
+        possibleTargetElem.nodeType === Node.ELEMENT_NODE &&
+        isInteractiveElement(possibleTargetElem)
```

Comment:
With the suggestion below, I think this will become just:

```suggestion
        isInteractiveElement(possibleTargetElem)
```

Reply:
Yep! One addition I missed previously: I have to do a type guard to with `instanceof Node` to make the call type check. As far as I can tell that's the only way to refine the [EventTarget interface](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget) into `Node` in a type safe way (I could be wrong about that, I'm no TypeScript expert). My previous code with the `as` assertion wasn't actually safe, whoops 😬 .

The `instanceof` usage does lose the cross-frame capability, though. Not sure if that matters for particular this listener?